src/composables/useActiveSession.js

```diff
@@ -0,0 +1,132 @@
+/*
+ * @copyright Copyright (c) 2023 Maksim Sukharev <antreesy.web@gmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { computed, onBeforeMount, onBeforeUnmount, ref, watch } from 'vue'
+
+import { getCapabilities } from '@nextcloud/capabilities'
+import { showInfo } from '@nextcloud/dialogs'
+
+import { SESSION } from '../constants.js'
+import { setSessionState } from '../services/participantsService.js'
+import { useStore } from './useStore.js'
+
+const supportSessionState = getCapabilities()?.spreed?.features?.includes('session-state')
+const INACTIVE_TIME_MS = 3 * 60 * 1000
+/**
+ * Check whether the current session is active or not:
+ * - tab or browser window was moved to background or minimized
+ * - there was not movements within tab window for a long time
+ *
+ * @return {boolean|undefined}
+ */
+export function useActiveSession() {
+	if (!supportSessionState) {
+		return supportSessionState
+	}
+
+	const store = useStore()
+	const token = computed(() => store.getters.getToken())
+	const windowIsVisible = computed(() => store.getters.windowIsVisible())
+
+	const inactiveTimer = ref(null)
+	const currentState = ref(SESSION.STATE.ACTIVE)
+
+	watch(token, () => {
+		// Joined conversation has active state by default
+		currentState.value = SESSION.STATE.ACTIVE
+	})
+
+	watch(windowIsVisible, (value) => {
+		// Change state if tab is hidden or minimized
+		// FIXME upstream: Notifications app doesn't work in background
```

Comment:
Not sure what you mean, the notifications app should work fine.
It even has code to track visibility because we change the page title to prefix it with `*` when you receive something, so pinned tabs are highlighted.

Reply:
I don't see native browser notifications, when tab is minimized, only when tab is opened again, new notification is added to the list and trigger browser notification to show